src/collectors/mattermost/mattermost.py

```diff
@@ -0,0 +1,242 @@
+# coding=utf-8
+
+"""
+Mattermost (https://www.mattermost.org/) is an open source, self-hosted
+Slack-alternative. Out of the box it comes with only support for the number
+of users and messages on limited historical data.
+
+This collector aims at extracting the data from the Mattermost database, so
+you can store it in your favorite metrics backend and do your own processing.
+
+The collector lets you define in what detail you want the counters to be. Be
+aware that for large numbers of groups/channels/users, the number of counters
+may grow quickly. Also since the collection performs GROUP BY queries for these
+cases, running the collection too frequently may have an impact on the
+performance of your system.
+
+
+Mattermost supports both postgresql and mysql. At this moment, the collector
+only support postgresql.
+
+#### Dependencies
+
+ * Psycopg2 : for postgresql connectivity
+
+"""
+
+import diamond.collector
+import psycopg2
+
+
+class MattermostCollector(diamond.collector.Collector):
+
+    RETRY = 5
+
+    def __init__(self, *args, **kwargs):
+        super(MattermostCollector, self).__init__(*args, **kwargs)
+        self.conn = None
+
+    def get_default_config_help(self):
+        config_help = super(MattermostCollector,
+                            self).get_default_config_help()
+        config_help.update({
+            'databaseHost':   'location of the server',
+            'databasePort':   'port of the server',
+            'databaseUser':   'username of the database',
+            'databasePwd':    'password of the user',
+            'collectTeamDetails':  'calculate details per team',
+            'collectChannelDetails': 'calculate details per channel',
+            'collectUserDetails':   'calculate details per user',
+        })
+        return config_help
+
+    def get_default_config(self):
+        """
+        Returns the default collector settings
+        """
+        config = super(MattermostCollector, self).get_default_config()
+        config.update({
+            'path':     'mattermost',
+            'databaseHost':   'localhost',
+            'databasePort':   '5432',
+            'databaseUser':   'mmuser',
+            'databasePwd':    'mmuser_password',
```

Comment:
since these are specified on the installation page (https://docs.mattermost.com/install/prod-ubuntu.html) I think it is not wrong to have them in... they're not exposing any secret, and this way, the collector works out-of-the-box with a typical install..